lib/nuget-parser/dotnet-core-parser.ts

```diff
@@ -45,61 +46,112 @@ function collectFlatList(targetObj) {
   });
 }
 
-function buildTreeRecursive(targetDeps, depName, parent, treeDepth) {
-  const MAX_TREE_DEPTH = 40;
-  if (treeDepth > MAX_TREE_DEPTH) {
-    throw new BigTreeError('The depth of the tree is too big.');
+interface DepLink {
+  from: string;
```

Comment:
instead of `string` can you make them type `Dependency` so we don't have to do all the splits later on?
You can make the Dependency.dependencies field optional